### ISSUE TITLE
Pihole upgrade fails to start

### ISSUE THREAD

**Issue by nicholas-c** [unifi-utilities/unifios-utilities#425]
**Describe the bug**
Installed Pihole with DoH around 16-17 months ago, needed to update the image, so followed the guide and created the `udp_update.sh` script. Contain then fails to start afterwards.

Working backwards through the available images, `boostchicken/pihole:2022.07.1` is the last image to work, any later images cause this issue

```
s6-rc: info: service s6rc-oneshot-runner: starting
s6-rc: info: service s6rc-oneshot-runner successfully started
s6-rc: info: service fix-attrs: starting
s6-rc: info: service fix-attrs successfully started
s6-rc: info: service legacy-cont-init: starting
cont-init: info: running /etc/cont-init.d/10-cloudflared.sh
/package/admin/s6-overlay-3.1.1.2/etc/s6-rc/scripts/cont-init: 14: /etc/cont-init.d/10-cloudflared.sh: Permission denied
cont-init: info: /etc/cont-init.d/10-cloudflared.sh exited 126
cont-init: warning: some scripts exited nonzero
s6-rc: warning: unable to start service legacy-cont-init: command exited 1
/run/s6/basedir/scripts/rc.init: warning: s6-rc failed to properly bring all the services up! Check your logs (in /run/uncaught-logs/current if you have in-container logging) for more information.
prog: fatal: stopping the container.
s6-rc: info: service fix-attrs: stopping
s6-rc: info: service fix-attrs successfully stopped
s6-rc: info: service s6rc-oneshot-runner: stopping
s6-rc: info: service s6rc-oneshot-runner successfully stopped
```

**To Reproduce**
1. Have older installation of run-pihole
2. Create `udp_update.sh` script as per latest docs
3. Run script and have latest docker images pulled
4. See persistent error

**Expected behavior**
Expect the container to start

**Screenshots**
<img width="1191" alt="image" src="https://user-images.githubusercontent.com/6815423/194487076-1dd4250c-2c68-4015-a5a7-98de9254431e.png">

**UDM Information**
 - Variant: UDM Pro
 - Firmware Version:: UniFi OS UDM Pro 1.12.30
 - Controller Version: Network 7.2.94

**Comment by charliemac93** [comment]
any updates on this? i'm experiencing the exact same issue since tag 2022.08.3 and had to revert to 2022.07.1. 

Variant: UDM Pro
Firmware Version:: UniFi OS UDM Pro 1.12.30
Controller Version: Network 7.3.36

**Comment by jaromirrivera** [comment]
This looks like a result of the https://github.com/pi-hole/docker-pi-hole repo reorganizing some files from 2022.07.1 to 2022.8.3 and above.

In 2022.8.3 and above there is no /etc/conf-init.d folder... looks like the repo changed the way startup processes work...

**Comment by emlynmac** [comment]
I just had a look at the overlay updates and if you want to make it work, you just need to make that directory. The directory is used in v3 as per the v2.
I run DOTE, not cloudflared (any more) so the filename is different, but the creation of the parent directory is the same.

```
RUN chmod +x /opt/dote && mkdir -p /etc/cont-init.d && echo -e "#!/bin/sh\n/opt/dote -s 127.0.0.1:5053 -d\n" > /etc/cont-init.d/10-dote.sh && chmod +x /etc/cont-init.d/10-dote.sh
```

**Comment by boostchicken** [comment]
This is now fixed and on 2023.1
